services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyHttpHandlerHelper.java

```diff
@@ -23,20 +23,34 @@
 
 public class ConsoleProxyHttpHandlerHelper {
     private static final Logger s_logger = Logger.getLogger(ConsoleProxyHttpHandlerHelper.class);
+    private static final String AND = "&";
+    private static final String EQUALS = "=";
+    private static final String HOST = "host";
+    private static final String PORT = "port";
+    private static final String SID = "sid";
+    private static final String TAG = "tag";
+    private static final String CONSOLE_URL = "consoleurl";
+    private static final String SESSION_REF = "sessionref";
+    private static final String TICKET = "ticket";
+    private static final String LOCALE = "locale";
+    private static final String HYPERV_HOST = "hypervHost";
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+    private static final String TOKEN = "token";
 
     public static Map<String, String> getQueryMap(String query) {
-        String[] params = query.split("&");
```

Comment:
Is there a way to parse URL (possibly a get request) to get the key/value map of params?

Reply:
Yes, there is, thanks for pointing that!
I will ping you when update it (also extracting some lines to methods, bringing some test cases, and documenting).